sdks/python/apache_beam/io/gcp/bigquery_read_perf_test.py

```diff
@@ -126,9 +133,21 @@ def format_record(record):
     p.run().wait_until_finish()
 
   def test(self):
+    def extract_values(row):
+      """Extracts value from a row."""
+      yield base64.b64decode(row.values()[0])
+
     self.result = (self.pipeline
                    | 'Read from BigQuery' >> Read(BigQuerySource(
                        dataset=self.input_dataset, table=self.input_table))
+                   | 'Measure bytes' >> ParDo(MeasureBytes(
+                       self.metrics_namespace, extract_values))
+                   | 'Count messages' >> ParDo(CountMessages(
+                       self.metrics_namespace))
+                   | 'Measure time: Start' >> ParDo(MeasureTime(
+                       self.metrics_namespace))
+                   | 'Measure time: End' >> ParDo(MeasureTime(
```

Comment:
> I don't understand what the difference is between Start and End versions of MeasureTime. 
Indeed, there is no difference between Start and End versions in this case. I'll leave only one monitor then.

> What time values do you extract from these metrics?
In general, MeasureTime steps checks the processing time of the first element (let's say: `x`) and the last element (let's say: `y`) in PCollection. The difference between the minimum of `x` and the maximum of `y` is in approximation total running time. I believe that's the best way to measure IOs performance for now, since we don't have any metrics gathered by the IO itself.